rest_framework/test.py

```diff
@@ -22,7 +23,7 @@
 
 
 def force_authenticate(request, user=None, token=None):
-    request._force_auth_user = user
+    request._force_auth_user = copy.deepcopy(user)
```

Comment:
I'm a little concerned about the use of deepcopy here. It's pretty much magic, and there's no guarantee that it might not fail in some cases (user can be a custom model, or have been modified by middleware, and might have almost anything stored in state on it)

I'm not sure what the best action is here. Perhaps we should catch exceptions and fall back to simply assigning `user`, tho that seems problematic too.

Perhaps I'm being overly cautious about the possibility of `deepcopy` failing here?